apps/api/src/app/integrations/usecases/get-webhook-support-status/get-webhook-support-status.usecase.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { IntegrationEntity, IntegrationRepository } from '@novu/dal';
+import { IEmailProvider, ISmsProvider } from '@novu/stateless';
+import { MailFactory, SmsFactory, ISmsHandler, IMailHandler } from '@novu/application-generic';
+import { ChannelTypeEnum } from '@novu/shared';
+import { GetWebhookSupportStatusCommand } from './get-webhook-support-status.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+@Injectable()
+export class GetWebhookSupportStatus {
+  public readonly mailFactory = new MailFactory();
+  public readonly smsFactory = new SmsFactory();
+  private provider: IEmailProvider | ISmsProvider;
+
+  constructor(private integrationRepository: IntegrationRepository) {}
+
+  async execute(command: GetWebhookSupportStatusCommand): Promise<boolean> {
+    const { providerId } = command;
+
+    const integration: IntegrationEntity = await this.getIntegration(command);
+    const { channel } = integration;
+
+    if (!integration) {
+      throw new NotFoundException(`Integration for ${providerId} was not found`);
+    }
```

Comment:
I think we need to invert the order here. If we are going to check for the existence of integration and break the flow makes me think we should try to access the channel property after doing that check. Also if `integration` is `null` or `undefined` accessing to it will throw an uncontrolled TypeError.